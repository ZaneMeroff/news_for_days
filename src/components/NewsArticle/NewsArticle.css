* {
  margin: 0px;
  color: white;
}

.news-article {
  border: 2px solid red;
  height: auto;
  width: auto;
}

.news-image {
  height: 200px;
  width: 200px;
}
