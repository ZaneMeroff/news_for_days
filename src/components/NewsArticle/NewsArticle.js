import React from 'react';
import './NewsArticle.css';

const NewsArticle = props => {
  return (
    <div className='news-article'>
      <img className='news-image' src={props.image}/>
      <h1 className='news-headline'>{props.headline}</h1>
      <p className='news-description'>{props.description}</p>
      <button className='read-article-button'>read article</button>
    </div>
  )
}

export default NewsArticle;
