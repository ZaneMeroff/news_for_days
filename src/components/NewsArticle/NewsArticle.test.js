import React from 'react';
import NewsArticle from '../NewsArticle/NewsArticle.js';
import { shallow } from 'enzyme';

describe('NewsArticle', () => {

  it('should be able to render news article to page', () => {
    const wrapper = shallow(<NewsArticle
      image={'https://static01.nyt.com/images/2019/09/27/multimedia/27xp-spiderman/27xp-spiderman-superJumbo.jpg?quality=90&auto=webp'}
      headline={'Spider-Man Will Remain in the Marvel Cinematic Universe'}
      description={'The web slinger will be the only superhero with the ability to cross between Disney and Sonyâ€™s cinematic universes, according to Marvel Studios.'}
      url={'https://www.nytimes.com/2019/09/27/business/media/spiderman-mcu-sony-marvel.html'}
      key={1}
      />)
    expect(wrapper).toMatchSnapshot();
  });

});
