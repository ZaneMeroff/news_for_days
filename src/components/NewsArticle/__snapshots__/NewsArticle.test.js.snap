// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should be able to render news article to page 1`] = `
<div
  className="news-article"
>
  <img
    className="news-image"
    src="https://static01.nyt.com/images/2019/09/27/multimedia/27xp-spiderman/27xp-spiderman-superJumbo.jpg?quality=90&auto=webp"
  />
  <h1
    className="news-headline"
  >
    Spider-Man Will Remain in the Marvel Cinematic Universe
  </h1>
  <p
    className="news-description"
  >
    The web slinger will be the only superhero with the ability to cross between Disney and Sonyâ€™s cinematic universes, according to Marvel Studios.
  </p>
  <button
    className="read-article-button"
  >
    <a
      href="https://www.nytimes.com/2019/09/27/business/media/spiderman-mcu-sony-marvel.html"
      target="_blank"
    >
      read article
    </a>
  </button>
</div>
`;
